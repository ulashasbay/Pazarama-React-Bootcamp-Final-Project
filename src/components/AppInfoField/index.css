.row-app-show-page {
  /* display: flex; */
  /* flex-direction: column; */
  margin-bottom: 4px;
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  font-size: 18px;
}

.app-info-area-img {
  width: 100px;
  margin-left: 36%;
  border: solid 1px rgb(221, 220, 220);
}
