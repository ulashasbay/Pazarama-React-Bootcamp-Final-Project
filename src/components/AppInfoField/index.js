import React from "react";
import { useSelector } from "react-redux";
import "./index.css";

function AppInfoField() {
  const appIdValue = useSelector((state) => state.appId.value);
  const userInfoValue = useSelector((state) => state.userInfo.value);

  return (
    <>
      {"imageUrl" in userInfoValue && (
        <img
          className="app-info-area-img"
          width="100px"
          src={userInfoValue.imageUrl}
          alt="img"
        />
      )}
      <div className="row-app-show-page">
        <b>Başvuru No:</b>
        {appIdValue}
      </div>
      <div className="row-app-show-page">
        <b>T.C. Kimlik No:</b>
        {userInfoValue.tcNo}
      </div>
      <div className="row-app-show-page">
        <b>Ad:</b>
        {userInfoValue.name}
      </div>
      <div className="row-app-show-page">
        <b>Soyad:</b>
        {userInfoValue.surname}
      </div>
      <div className="row-app-show-page">
        <b>Yaş:</b>
        {userInfoValue.age}
      </div>
      <div className="row-app-show-page">
        <b>Adres:</b>
        {userInfoValue.address}
      </div>
      <div className="row-app-show-page">
        <b>Başvuru Nedeni:</b>
        {userInfoValue.applicationReason}
      </div>
      <div className="row-app-show-page">
        <b>Başvuru Durumu:</b>
        {userInfoValue.applicationStatus}
      </div>
      <div className="row-app-show-page">
        <b>Başvuru Sonucu:</b>
        {userInfoValue.applicationResult}
      </div>
    </>
  );
}

export default AppInfoField;
