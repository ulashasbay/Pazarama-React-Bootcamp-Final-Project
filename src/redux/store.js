import { configureStore } from "@reduxjs/toolkit";
import appIdReducer from "./appId/appIdSlice";
import userInfoReducer from "./userInfo/userInfoSlice";
import isLoggedInReducer from "./isLoggedIn/isLoggedInSlice";

export const store = configureStore({
  reducer: {
    appId: appIdReducer,
    userInfo: userInfoReducer,
    isLoggedIn: isLoggedInReducer,
  },
});
