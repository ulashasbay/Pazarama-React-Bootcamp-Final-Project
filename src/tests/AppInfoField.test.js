import React from "react";
import { render, screen } from "@testing-library/react";
import AppInfoField from "../components/AppInfoField";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { store } from "../redux/store";
import renderer from "react-test-renderer";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <Provider store={store}>{children}</Provider>
    </BrowserRouter>
  );
};

describe("AppInfoField Test", () => {
  beforeEach(() => {
    render(<AppInfoField />, { wrapper: AllTheProviders });
  });

  test("Render AppInfoField items", () => {
    const basvuruNo = screen.getByText("Başvuru No:");
    expect(basvuruNo).toBeInTheDocument();

    const ad = screen.getByText("Ad:");
    expect(ad).toBeInTheDocument();

    const soyad = screen.getByText("Soyad:");
    expect(soyad).toBeInTheDocument();

    const yas = screen.getByText("Yaş:");
    expect(yas).toBeInTheDocument();

    const tcNo = screen.getByText("T.C. Kimlik No:");
    expect(tcNo).toBeInTheDocument();

    const basvuruNedeni = screen.getByText("Başvuru Nedeni:");
    expect(basvuruNedeni).toBeInTheDocument();

    const adres = screen.getByText("Adres:");
    expect(adres).toBeInTheDocument();

    const basvuruSonucu = screen.getByText("Başvuru Sonucu:");
    expect(basvuruSonucu).toBeInTheDocument();
  });
});

test("AppInfoField renders correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Provider store={store}>
          <AppInfoField />
        </Provider>
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
