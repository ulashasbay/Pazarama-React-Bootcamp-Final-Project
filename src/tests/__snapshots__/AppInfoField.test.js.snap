// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppInfoField renders correctly 1`] = `
Array [
  <div
    className="row-app-show-page"
  >
    <b>
      Başvuru No:
    </b>
    
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      T.C. Kimlik No:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Ad:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Soyad:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Yaş:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Adres:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Başvuru Nedeni:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Başvuru Durumu:
    </b>
  </div>,
  <div
    className="row-app-show-page"
  >
    <b>
      Başvuru Sonucu:
    </b>
  </div>,
]
`;
